fibers/posix-clocks.scm.in

```diff
@@ -113,15 +128,26 @@
      (+ (* sec #e1e9) nsec))))
 
 (define clock-nanosleep
-  (let* ((ptr (dynamic-pointer "clock_nanosleep" exe))
+  (if is-osx?
+    (let* ((ptr (dynamic-pointer "nanosleep" exe))
+         (proc (pointer->procedure int ptr (list '* '*))))
+    (lambda* (clockid nsec #:key absolute? (buf (nsec->timespec nsec)))
+      (let* ((flags (if absolute? TIMER_ABSTIME 0))
+             (ret (proc buf buf)))
+        (cond
+         ((zero? ret) (values #t 0))
+         ((eqv? ret EINTR) (values #f (timespec->nsec buf)))
```

Comment:
same for `timespec->nsec`. i'm unable to find this :-(

Reply:
Check here https://github.com/Habush/fibers/blob/3c00a8c65555bf93e08372bcbc59b3b42dd7cccc/fibers/posix-clocks.scm.in#L125

Reply:
Duhh.... my bad.I was looking at a modified version I had locally.